stock_inventory/models/stock_inventory.py

```diff
@@ -196,26 +196,53 @@ def refresh_stock_quant_ids(self):
             rec.stock_quant_ids = rec._get_quants(rec.location_ids)
 
     def action_state_to_in_progress(self):
+        # Get products based on product_selection
+        if self.product_selection == "all":
+            self_products = self.env["product.product"].search([])
+        elif self.product_selection == "category" and self.category_id:
+            domain = [
+                "|",
+                ("categ_id", "=", self.category_id.id),
+                ("categ_id", "in", self.category_id.child_id.ids),
+            ]
+            self_products = self.env["product.product"].search(domain)
+        else:
+            self_products = self.product_ids
         active_rec = self.env["stock.inventory"].search(
             [
                 ("state", "=", "in_progress"),
                 ("location_ids", "child_of", self.location_ids.ids),
-            ],
-            limit=1,
+            ]
         )
-        if active_rec:
+        overlapping_inventories = active_rec.filtered(
+            lambda x: any(
+                product in self_products
+                for product in (
+                    x.product_ids
+                    if x.product_selection != "all"
+                    else self.env["product.product"].search([])
+                    or self.env["product.product"].search(
+                        [("categ_id", "=", x.category_id.id)]
+                        if x.product_selection == "category" and x.category_id
+                        else []
+                    )
+                )
+            )
+        )
```

Comment:
it could be good to move the logic to get an overlapping inventory in a separate method and use it also for the constrain

Reply:
I agree, done that!